internal/ansible/flags/flag.go

```diff
@@ -160,12 +161,6 @@ func (f *Flags) AddTo(flagSet *pflag.FlagSet) {
 			" holding the leader lock (required if running locally with leader"+
 			" election enabled).",
 	)
-	flagSet.StringVar(&f.LeaderElectionResourceLock,
-		"leader-elect-resource-lock",
-		"configmapsleases",
-		"The type of resource object that is used for locking during leader election."+
-			" Supported options are 'leases', 'endpointsleases' and 'configmapsleases'. Default is configmapsleases.",
-	)
```

Comment:
I don't think we can just remove this without it being a breaking change and requiring a new major version. I'd prefer if we changed this to something like:
```go
	flagSet.StringVar(&f.LeaderElectionResourceLock,
		"leader-elect-resource-lock",
		resourcelock.LeasesResourceLock,
		"determines which resource lock to use for leader election, defaults to \"leases\". Change this value only if you know what you are doing.",
	)
```

and then remove the hardcoded setting below

Reply:
I think it's also acceptable to deprecate this flag and have it "do nothing". We just can't remove it right now

Reply:
This flag isn't in the `helm-plugin` so I was trying to remove it for consistency/clarity across plugins. I'll move to deprecate it, since IMO that's more clear to end users.